pandas/tests/frame/test_constructors.py

```diff
@@ -2080,6 +2080,14 @@ def test_constructor_series_copy(self, float_frame):
 
         assert not (series["A"] == 5).all()
 
+    def test_object_array_does_not_copy(self):
+        a = np.array(["a", "b"], dtype="object")
+        b = np.array([["a", "b"], ["c", "d"]], dtype="object")
+        df = DataFrame(a)
+        assert np.shares_memory(df.values, a)
+        df2 = DataFrame(b)
+        assert np.shares_memory(df2.values, b)
+
```

Comment:
assert the frames are equal as well

Reply:
did you mean assert the numpy arrays equal? They're not equal, but they do share memory:
```
>       np.testing.assert_array_equal(df.values, a)
E       AssertionError: 
E       Arrays are not equal
E       
E       (shapes (2, 1), (2,) mismatch)
E        x: array([['a'],
E                  ['b']], dtype=object)
E        y: array(['a', 'b'], dtype=object)
```
(note also I've split the test into two)